src/reconcile/rec_write.c

```diff
@@ -576,17 +600,29 @@ __rec_write_check_complete(
 		return (EBUSY);
 
 	/*
-	 * Eviction can configure lookaside table reconciliation, consider if
-	 * it's worth giving up this reconciliation attempt and falling back to
-	 * using the lookaside table.  We continue with evict/restore if
-	 * switching to the lookaside doesn't make sense for any reason: we
-	 * won't retry an evict/restore reconciliation until/unless the
-	 * transactional system moves forward, so at worst it's a single wasted
-	 * effort.
+	 * If no entries were used, the page is empty and we can only restore
+	 * updates against an empty row store leaf page.  (Column store modify
+	 * will attempt to allocate a zero-length array).
+	 */
+	if (r->page->type != WT_PAGE_ROW_LEAF) {
+		if (r->entries == 0 && r->supd_next != 0)
+			return (EBUSY);
+		for (multi = r->multi, i = 0; i < r->multi_next; ++multi, ++i)
+			if (multi->disk_image == NULL &&
+			    multi->supd_entries != 0)
+				return (EBUSY);
+	}
```

Comment:
The test against `r->entries == 0` concerns me. I'm not sure it's wrong, but...

Do you remember the test case that required this change?

My inclination would be to move this check into `__rec_split_write()`, rather than waiting until the final check. (It then becomes clear the check would be against `WT_CHUNK.entries`, too, which resolves my first question.)